ddtrace/tracer.py

```diff
@@ -145,7 +145,7 @@ def _default_span_processors_factory(
     # FIXME: type should be AppsecSpanProcessor but we have a cyclic import here
     """Construct the default list of span processors to use."""
     trace_processors = []  # type: List[TraceProcessor]
-    trace_processors += [TraceTagsProcessor()]
+    trace_processors += [TraceTagsProcessor(), peer_service_processor]
```

Comment:
Can we remove the `_ peer_service_processor` property from the tracer and just define it here?

```suggestion
    trace_processors += [TraceTagsProcessor(), PeerServiceProcessor(PeerServiceConfig())]
```

Reply:
1. Sure! The code has three patterns (pass in a processor, pass in a processor class, and insantiate without passing in), so I just picked one.  If we prefer this pattern, that's fine by me.

2. Already updated, it is now `process_trace` instead of `on_finish`

3. Is the performance improvement worth the loss of clarity?  The TraceTagsProcessor adds trace-level tags, which is different than adding span-level tags for all spans in a trace.  We can combine them and do both, but I don't think the performance change in this case is significant enough to warrant a refactor.  Open to debate or override, though!

Reply:
We can hold off on 3 for now